src/openslide-vendor-dicom.c

```diff
@@ -698,6 +700,100 @@ static bool maybe_add_file(openslide_t *osr,
   }
 }
 
+struct property_iterate {
+  openslide_t *osr;
+  const char *prefix;
+};
+
+static bool add_properties_element(const DcmElement *element,
+                                   void *client);
+
+static bool add_properties_dataset(const DcmDataSet *dataset,
+                                   void *client) {
+  return dcm_dataset_foreach(dataset, add_properties_element, client);
+}
+
+static bool add_properties_element(const DcmElement *element,
+                                   void *client) {
+  struct property_iterate *iter = (struct property_iterate *) client;
+  DcmVR vr = dcm_element_get_vr(element);
+  uint32_t tag = dcm_element_get_tag(element);
+  const char *keyword = dcm_dict_keyword_from_tag(tag);
+  DcmVRClass klass = dcm_dict_vr_class(vr);
+
+  // ignore unknown tags
+  if (!keyword) {
+    return true;
+  }
+
+  if (klass == DCM_CLASS_SEQUENCE) {
+    DcmSequence *seq;
+    if (dcm_element_get_value_sequence(NULL, element, &seq)) {
+      char *new_prefix = g_strdup_printf("%s%s.", iter->prefix, keyword);
+      struct property_iterate new_iter = { iter->osr, new_prefix };
+      dcm_sequence_foreach(seq, add_properties_dataset, &new_iter);
+      g_free(new_prefix);
+    }
+  } else {
+    char *value = g_strdup("");
```

Comment:
Without the inline lists, we should be able to skip the preallocation here.  If we don't recognize the type, I suspect we should skip adding the property rather than adding with an empty value.

Reply:
I swapped to array indexing for multivalues properties, so I think this is resolved.